share/jupyterhub/templates/home.html

```diff
@@ -2,6 +2,13 @@
 
 {% block main %}
 
+{% if announcement_home or announcement %}
```

Comment:
Nice idea.  I was able to simplify a bit more even and it seems to work (pending tests).  Another advantage is that there is now an `announcement` block which can be inherited separately in other templates - so that option (0) using your own templates becomes even easier.  Option (0) should also allow you to easily change the message without restarting!